File: Sort/C++/bubbleSort.cpp
#include<iostream>
using namespace std;

void bubbleSort(int a[],int n){
    for(int i=0;i<n;i++){
        for(int j=0;j<n-i-1;j++){
            if(a[j]>a[j+1]){
                int temp=a[j];
                a[j]=a[j+1];
                a[j+1]=temp;
            }
        }
    }
}

int main(){
    int a[5]={4,3,2,1,0};
    bubbleSort(a,5); //bubbleSort(array,size)
    for(int i=0;i<5;i++){
        cout<<a[i]<<"\n";
    }
    return 0;
}

File: Sort/C#/BubbleSort.cs
using System;

namespace BubbleSort
{
    class BubbleSort
    {
        public static void Main()
        {
            int[] testA = { 4, 3, 2, 1 };
            int[] testB = { 5, 5, 4, 1, 7 };

            PrintArray("testA", testA, false);
            PrintArray("sortedA", Sort(testA), true);

            PrintArray("testB", testB, false);
            PrintArray("sortedB", Sort(testB), true);
        }

        public static int[] Sort(int[] A)
        {
            for (int i = 0; i < A.Length; i++)
            {
                for (int j = A.Length - 1; j > i; j--)
                {
                    if (A[j] < A[j - 1])
                    {
                        // Swaps the two elements
                        var temp = A[j];
                        A[j] = A[j - 1];
                        A[j - 1] = temp;
                    }
                }
            }
            return A;
        }

        static void PrintArray(string name, int[] arr, bool isSorted)
        {
            Console.Write($"Array: {name}\nIs sorted: {isSorted}\n[");
            for (int i = 0; i < arr.Length; i++)
            {
                if (i == arr.Length - 1)
                {
                    Console.Write($" {arr[i]} ]\n\n");
                }
                else
                {
                    Console.Write($" {arr[i]}, ");
                }
            }
        }

    }
}
